skbio/metadata/_intersection.pyx

```diff
@@ -417,6 +426,12 @@ cdef class IntervalTree:
 
     # ---- Position based interfaces -----------------------------------------
 
+    def get_right_bound( self ):
```

Comment:
I think it should be removed. The method isn't necessary for this PR, and adding unused functionality increases maintenance burden.